### PR TITLE
deps(otel): Update OTel to v0.58.0

### PR DESCRIPTION
**Pull request by BinaryFissionGames**
### Proposed Change
* Update OTel deps to v0.58.0
* Remove awsprometheusremotewriteexporter
  * This exporter was previously deprecated, and removed in v0.58.0 of contrib
* Replace WithMetricsExporterAndStabilityLevel and similar with WithMetricsExporter or its equivalent

Note: This PR does not contain updates to operations collector dependencies.
Tested that this builds and runs locally.

##### Checklist
- [x] Changes are tested
- [ ] CI has passed

### CODE REVIEW

**Comment by schmikei** [comment]
Note that this has some postgres changes but they are hidden behind feature gates 

https://github.com/open-telemetry/opentelemetry-collector-contrib/releases/tag/v0.58.0

File: factories/exporters.go [hunk 1]
```diff
@@ -54,7 +53,6 @@ var defaultExporters = []component.ExporterFactory{
 	awscloudwatchlogsexporter.NewFactory(),
 	awsemfexporter.NewFactory(),
 	awskinesisexporter.NewFactory(),
-	awsprometheusremotewriteexporter.NewFactory(),
```
Comment (cpheps) [inline factories/exporters.go:57]: Did we mean to remove this?

Comment (BinaryFissionGames) [reply factories/exporters.go:57]: Yeah, as mentioned in the PR description, it was previously deprecated, and has now been deleted from contrib in v0.58.0.

Comment (BinaryFissionGames) [reply factories/exporters.go:57]: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/13173 for context

**Comment by BinaryFissionGames** [comment]
Looking into test failures...

File: factories/exporters.go (hunk 1 above)
Comment (cpheps) [reply factories/exporters.go:57]: Ok I thought I still saw it in the documentation so that's why I asked.

File: docs/exporters.md [hunk 2]
```diff
@@ -4,33 +4,33 @@ Below is a list of supported exports with links to their documentation pages.
...
+| Alibaba Cloud Log Service Exporter                                | [alibabacloudlogserviceexporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.58.0/exporter/alibabacloudlogserviceexporter/README.md) |
+| AWS CloudWatch Logs Exporter                                      | [awscloudwatchlogsexporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.58.0/exporter/awscloudwatchlogsexporter/README.md) |
+| AWS CloudWatch EMF Exporter                                       | [awsemfexporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.58.0/exporter/awsemfexporter/README.md) |
+| AWS Kinesis Exporter                                              | [awskinesisexporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.58.0/exporter/awskinesisexporter/README.md) |
+| AWS Prometheus Remote Write Exporter                              | [awsprometheusremotewriteexporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.58.0/exporter/awsprometheusremotewriteexporter/README.md) |
```
Comment (cpheps) [inline docs/exporters.md:11]: Remove this please if we removed the exporter.

File: factories/exporters.go (hunk 1 above)
Comment (cpheps) [reply factories/exporters.go:57]: Added a comment to just point out the line in the exporter readme.

File: docs/exporters.md (hunk 2 above)
Comment (BinaryFissionGames) [reply docs/exporters.md]: good catch, removed.

**Review by cpheps — APPROVED** [review]

**Review by schmikei — APPROVED** [review]

**Pull request merged**